pallets/time-release/src/migration/v2.rs

```diff
@@ -0,0 +1,207 @@
+								schedule.total_amount()
+							},
+						)
+						.fold(Zero::zero(), |acc: BalanceOf<T>, amount| {
+							acc.saturating_add(amount.unwrap_or(Zero::zero()))
+						});
+					// Translate the lock to a freeze
+					MigrationToV3::<T, OldCurrency>::translate_lock_to_freeze(
+						account_id,
+						total_amount.into(),
+					);
+					maybe_count += 1;
+					log::info!(target: LOG_TARGET,"migrated {:?}", maybe_count);
+				},
+			);
+
+			StorageVersion::new(2).put::<Pallet<T>>(); // 1 w
+			let reads = (maybe_count * 2 + 1) as u64;
+			// REVIEW: Are we doing 2 writes per account?
+			let writes = (maybe_count * 2 + 1) as u64;
+			log::info!(target: LOG_TARGET, "🔄 Time Release migration finished");
+			let weight = T::DbWeight::get().reads_writes(reads, writes);
+			log::info!(target: LOG_TARGET, "Time Release Migration calculated weight = {:?}", weight);
+			weight
+		} else {
+			// storage was already migrated.
+			log::info!(target: LOG_TARGET, "Old Time Release Locks->Freezes migration attempted to run. Please remove");
+			T::DbWeight::get().reads(1)
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+		use frame_support::storage::generator::StorageMap;
+		let pallet_prefix = v1::ReleaseSchedules::<T>::module_prefix();
+		let storage_prefix = v1::ReleaseSchedules::<T>::storage_prefix();
+		assert_eq!(&b"TimeRelease"[..], pallet_prefix);
+		assert_eq!(&b"ReleaseSchedules"[..], storage_prefix);
+		log::info!(target: LOG_TARGET, "Running pre_upgrade...");
+
+		let count = v1::ReleaseSchedules::<T>::iter().count() as u32;
+		log::info!(target: LOG_TARGET, "Finish pre_upgrade for {:?} records", count);
+		Ok(count.encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+		use crate::ReleaseSchedules;
+		use parity_scale_codec::Decode;
+		let pre_upgrade_count: u32 = Decode::decode(&mut state.as_slice()).unwrap_or_default();
+		let on_chain_version = Pallet::<T>::on_chain_storage_version();
+
+		assert_eq!(on_chain_version, crate::module::STORAGE_VERSION);
+		assert_eq!(pre_upgrade_count as usize, ReleaseSchedules::<T>::iter().count());
+
+		log::info!(target: LOG_TARGET, "✅ migration post_upgrade checks passed");
+		Ok(())
+	}
+}
+
+#[cfg(test)]
+#[cfg(feature = "try-runtime")]
+mod test {
+	use frame_support::traits::{tokens::fungible::InspectFreeze, WithdrawReasons};
+
+	use super::*;
+	use crate::mock::{Test, *};
+	use pallet_balances::{BalanceLock, Reasons};
+
+	type MigrationOf<T> = MigrationToV3<T, pallet_balances::Pallet<T>>;
+
+	#[test]
+	fn migration_works() {
+		ExtBuilder::build().execute_with(|| {
+			StorageVersion::new(2).put::<Pallet<Test>>();
+			// Create some data in the old format
+			// Grab an account with a balance
+			let account = 200;
+			let amount = 50;
+
+			pallet_balances::Pallet::<Test>::set_lock(
```

Comment:
Don't you need to add something into v1::ReleaseSchedules first for the test since that is the main storage that you are iterating on?